frontend/src/metabase/admin/permissions/utils/graph/partial-updates.ts

```diff
@@ -0,0 +1,97 @@
+import _ from "underscore";
+
+import type { GroupsPermissions } from "metabase-types/api";
+
+type AdvancedPermissions = Record<
+  string,
+  { group_id: number; [key: string]: unknown }[] | undefined
+>;
+
+interface AdvancedPermissionsData {
+  modifiedGroupIds: string[];
+  permissions: AdvancedPermissions;
+}
+
+// utils for dealing with partial graph updates
+
+function getModifiedGroupIdsInGraph(
+  groupIds: string[],
+  originalDataPermissions: GroupsPermissions,
+  newDataPermissions: GroupsPermissions,
+) {
+  // find only the groupIds that have had some kind of modification made in their permissions graph
+  return groupIds.filter(groupId => {
+    return !_.isEqual(
+      newDataPermissions[groupId as unknown as number],
```

Comment:
if this is really a string, shouldn't we 
```suggestion
      newDataPermissions[Number(groupId)],
```

?

Reply:
yeah, we constructed the permissions graph interface to be keyed on `GroupId` which is a `number`. this doesn't work as expected since javascript converts numbers to strings for object keys. I solved this by making a change to the type.